<launch>

  <arg name="left_gripper" default="true" />

  <!-- left gripper -->
  <group ns="gripper_front/limb/left/dxl" if="$(arg left_gripper)">
    <!-- Start the Dynamixel servo controller manager -->
    <node name="controller_manager"
          pkg="dynamixel_controllers" type="controller_manager.py"
          respawn="true" output="screen">
      <!-- Cannot run at more than 27 Hz-->
      <rosparam>
        namespace: dxl_manager
        serial_ports:
          port:
            port_name: "/dev/dxhub"
            baud_rate: 57600
            min_motor_id: 1
            max_motor_id: 4
            update_rate: 25
      </rosparam>
    </node>

    <!-- Start the Dynamixel servo controllers -->
    <rosparam file="$(find sphand_driver)/config/left_gripper_v8/dxl_controllers.yaml" command="load"/>
    <node name="controller_spawner"
          pkg="dynamixel_controllers" type="controller_spawner.py"
          args="--manager=dxl_manager
                --port port
                prismatic_joint_controller
                vacuum_pad_tendon_controller
                finger_yaw_joint_controller
                finger_tendon_controller"
          output="screen" />
  </group>

  <!-- right gripper -->
  <group ns="gripper_front/limb/right/dxl" unless="$(arg left_gripper)">
    <!-- Start the Dynamixel servo controller manager -->
    <node name="controller_manager"
          pkg="dynamixel_controllers" type="controller_manager.py"
          respawn="true" output="screen">
      <rosparam>
        namespace: dxl_manager
        serial_ports:
          port:
            port_name: "/dev/dxhub"
            baud_rate: 57600
            min_motor_id: 1
            max_motor_id: 4
            update_rate: 20
      </rosparam>
    </node>

    <!-- Start the Dynamixel servo controllers -->
    <rosparam file="$(find sphand_driver)/config/right_gripper_v7/dxl_controllers.yaml" command="load"/>
    <node name="controller_spawner"
          pkg="dynamixel_controllers" type="controller_spawner.py"
          args="--manager=dxl_manager
                --port port
                prismatic_joint_controller
                vacuum_pad_tendon_controller
                finger_yaw_joint_controller
                finger_tendon_controller"
          output="screen" />
  </group>

</launch>
